docs/content/en/integrations/social-authentication.md

```diff
@@ -191,6 +191,26 @@ user, such as 'staff' or 'superuser'
 5.  Restart your Dojo, and you should now see a **Login with Azure AD**
     button on the login page which should *magically* work
 
+### Automatic Import of User-Groups
+To import groups from AzureAD users, the following env. var needs to be set:  
+
+    {{< highlight python >}}
+    DD_SOCIAL_AUTH_AZUREAD_TENANT_OAUTH2_GET_GROUPS=True
```

Comment:
the GET_GROUPS as variable name might be confusing. Does it get the groups from the Azure AD Graph API and store them in DD? Does it get the groups from the token an update the membership for the user in DD? Or Both?
In the code I can see it does both. Might be better to have 2 flags as some people actually might not want to call the graph api because the tokens involved don't allow querying the graph api for groups.

Reply:
Another scenario that I see people using as that they want users to be added to any groups specified in the AAD token, but they do not want new groups to be automatically created. So instead of regex filter, they don't want any new groups to be created. I know this can be achieved by using the regex filter to contain all the groups you have in DD, but that would mean the regex needs to be updated everytime a group is added/removed.

Reply:
From my point of view, it shouldn't be the target to have groupIDs in DefectDojo. Therefore, Azure AD should provide valid groups (which is not possible in all cases) or the graph should be able to be called. In case group IDs are returned and the graph api can not be called, what would be the value of the IDs in DefectDojo? Therefore, I do not recommend such a filter.

The second senario might be a case for an other issue. The best case would be UI driven with a database like in Dependency Track. I recommend to make one step at a time.

Reply:
yes, some things could be a future PR. Maybe we should suggest in the docs to use the application roles feature of Azure AD, I believe they result in human readable names in the tokens.